scripts/common.py

```diff
@@ -712,3 +715,91 @@ def __enter__(self):
+            for proc in procs:
+                proc.terminate()
+
+            _, procs_still_running = psutil.wait_procs(procs, timeout=10)
+            for proc in procs_still_running:
+                proc.kill()
+                logging.warn('Process killed (pid=%d)' % proc.pid)
+
+
+@contextlib.contextmanager
+def managed_dev_appserver(
+        app_yaml_path, env=None, log_level='info',
+        host='0.0.0.0', port=8080, admin_host='0.0.0.0', admin_port=8000,
+        clear_datastore=False, enable_console=False, enable_host_checking=True,
+        automatic_restart=True, skip_sdk_update_check=False):
+    """Returns a context manager to start up and shut down a GAE dev appserver.
+
+    Args:
+        app_yaml_path: str. Path to the app.yaml file which defines the
+            structure of the server.
+        env: dict(str: str) or None. Defines the environment variables for the
+            new process.
+        log_level: str. The lowest log level generated by the application code
+            and the development server. Expected values are: debug, info,
+            warning, error, critical.
+        host: str. The host name to which the app server should bind.
+        port: int. The lowest port to which application modules should bind.
+        admin_host: str. The host name to which the admin server should bind.
+        admin_port: int. The port to which the admin server should bind.
+        clear_datastore: bool. Whether to clear the datastore on startup.
+        enable_console: bool. Whether to enable interactive console in admin
+            view.
+        enable_host_checking: bool. Whether to enforce HTTP Host checking for
+            application modules, API server, and admin server. Host checking
+            protects against DNS rebinding attacks, so only disable after
+            understanding the security implications.
+        automatic_restart: bool. Whether to restart instances automatically when
+            files relevant to their module are changed.
+        skip_sdk_update_check: bool. Whether to skip checking for SDK updates.
+            If false, uses .appcfg_nag to decide.
+
+    Yields:
+        psutil.Process. The dev_appserver process.
+    """
+    dev_appserver_args = [
+        CURRENT_PYTHON_BIN, DEV_APPSERVER_PATH,
+        '--host', host, '--port', port,
+        '--admin_host', admin_host, '--admin_port', admin_port,
+        '--clear_datastore', 'true' if clear_datastore else 'false',
+        '--enable_console', 'true' if enable_console else 'false',
+        '--enable_host_checking', 'true' if enable_host_checking else 'false',
+        '--automatic_restart', 'true' if automatic_restart else 'false',
+        '--skip_sdk_update_check', 'true' if skip_sdk_update_check else 'false',
+        '--log_level', log_level, '--dev_appserver_log_level', log_level,
+        app_yaml_path
+    ]
+    with managed_process(dev_appserver_args, shell=True, env=env) as proc:
+        yield proc
```

Comment:
Could we do just?
```suggestion
    yield proc
```

Reply:
No, because the process will only be valid/maintained within the context manager.

The closest we could do is:
```suggestion
    return managed_process(dev_appserver_args, shell=True, env=env)
```

But the style would be different from the rest of our context managers, and @seanlip had concerns with that in the past.

Reply:
Hmm, okay, probably better to use a consistent style for this everywhere.